### PR TITLE
Migrate Notification components to TypeScript

### PR DESCRIPTION
**Pull request by victorhmp**
#### What problem is this solving?

Components under the `react/components/Notification` directory were not written in TypeScript yet, this PR solves that 🎉 .

#### How to test it?

[Workspace](https://victormiranda--storecomponents.myvtex.com/)

#### Depends on

- https://github.com/vtex-apps/native-types/pull/31 (build is going to fail until native-types is released)

### CODE REVIEW

**Review by kaisermann — APPROVED** [review]

File: react/components/Notification/NotificationInline.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  }
+
+  return <NotificationContent content={content} />
+}
+
+NotificationInline.schema = {
+  title: 'admin/editor.notification-bar.title',
+}
+
+export default memo(NotificationInline)
```
Comment (klzns) [inline react/components/Notification/NotificationInline.tsx:21]: Static properties are lost here.

You can do something like:

```tsx
const MemoNoticationInline = memo(NotificationInline)

MemoNotificationInline.schema = {
  title: 'admin/editor.notification-bar.title',
}

export default MemoNotificationInline
```

File: react/typings/vtex.native-types.d.ts [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
   import { ComponentType } from 'react'
+  import { MessageDescriptor } from 'react-intl'
 
   export const IOMessage: ComponentType<{
     id: string
     values: Record<string, any>
     ['data-testid']: string
   }>
+
+  export const formatIOMessage: (
```
Comment (klzns) [inline react/typings/vtex.native-types.d.ts:12]: Why not use the exported types from the app?

File: react/components/Notification/NotificationBar.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      </div>
+    </div>
+  )
+}
+
+NotificationBar.schema = {
+  title: 'admin/editor.notification-bar.title',
+}
+
+export default memo(NotificationBar)
```
Comment (klzns) [inline react/components/Notification/NotificationBar.tsx:42]: same

File: react/components/Notification/NotificationContent.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  const handles = useCssHandles(CSS_HANDLES)
+
+  return (
+    <div className={handles.notificationContent}>
+      <SanitizedHTML content={content} />
+    </div>
+  )
+}
+
+export default memo(NotificationContent)
```
Comment (klzns) [inline react/components/Notification/NotificationContent.tsx:22]: same

File: react/NotificationInline.js [hunk 5]
```diff
@@ -1,3 +1,3 @@
-import NotificationInline from './components/Notification/notificationInline'
+import NotificationInline from './components/Notification/NotificationInline'
```
Comment (klzns) [inline react/NotificationInline.js:1]: Can we have the code in this file instead?

**Review by klzns — CHANGES_REQUESTED** [review]

File: react/components/Notification/NotificationInline.tsx (hunk 1 above)
Comment (victorhmp) [reply react/components/Notification/NotificationInline.tsx:21]: Nice! Did not know that!

File: react/typings/vtex.native-types.d.ts (hunk 2 above)
Comment (victorhmp) [reply react/typings/vtex.native-types.d.ts:12]: I'll try it out, but I think they're not quite right... If that's the case, I'll open a PR to fix types for `vtex.native-types` and come back to this :)

File: react/NotificationInline.js (hunk 5 above)
Comment (victorhmp) [reply react/NotificationInline.js:1]: Sure! But then this file would need to be renamed... could that be a breaking change? 🤔

File: react/typings/vtex.native-types.d.ts (hunk 2 above)
Comment (victorhmp) [reply react/typings/vtex.native-types.d.ts:12]: Yep, the types are not quite there. I'll open a PR to fix them

Comment (victorhmp) [reply react/typings/vtex.native-types.d.ts:12]: There you go: https://github.com/vtex-apps/native-types/pull/31

File: react/package.json [hunk 6]
```diff
@@ -47,28 +47,28 @@
...
     "vtex.modal-layout": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.modal-layout@0.7.1/public/@types/vtex.modal-layout",
-    "vtex.native-types": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.native-types@0.7.1/public/@types/vtex.native-types",
-    "vtex.order-manager": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.order-manager@0.8.6/public/@types/vtex.order-manager",
-    "vtex.order-shipping": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.order-shipping@0.5.0/public/@types/vtex.order-shipping",
-    "vtex.product-context": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.product-context@0.9.0/public/@types/vtex.product-context",
-    "vtex.react-portal": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.react-portal@0.3.1/public/@types/vtex.react-portal",
-    "vtex.render-runtime": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.render-runtime@8.120.0/public/@types/vtex.render-runtime",
+    "vtex.native-types": "https://victormiranda--storecomponents.myvtex.com/_v/private/typings/linked/v1/vtex.native-types@0.7.3+build1606250594/public/@types/vtex.native-types",
```
Comment (victorhmp) [inline react/package.json:56]: This should be updated before release

**Review by klzns — APPROVED** [review]

**Review by icazevedo — APPROVED** [review]

**Pull request merged**